frontend/src/components/authbuttons/index.tsx

```diff
@@ -0,0 +1,70 @@
+import { Box, Button } from '@material-ui/core';
+import React from 'react';
+import { generatePath, useHistory } from 'react-router-dom';
+import { request } from '../../lib/k8s/apiProxy';
+import { getCluster, getClusterPrefixedPath } from '../../lib/util';
+import { ClusterDialog } from '../cluster/Chooser';
+import OauthPopup from '../oidcauth/OauthPopup';
+
+function AuthButtons(){
+  const history = useHistory();
+  const [authType, setAuthType] = React.useState<string | null>(null);
+  localStorage.setItem('selected_cluster', getCluster() as string);
+
+  function tokenAuthHandler(){
+    history.push({
+      pathname: generatePath(getClusterPrefixedPath('token'), {cluster: (getCluster() as string)}),
+      state: {
+        from: '/login'
+      }});
+  }
+
+  function oidcAuthHandler(){
+    history.replace({
+      pathname: generatePath(getClusterPrefixedPath(), {cluster: localStorage.getItem('selected_cluster') as string}),
+      state: {
+        from: '/login'
+      }
+    });
+  }
+  function oauthWindowCloseHandler(){
+    console.log('closed');
+  }
+  React.useEffect(() => {
+    request(`/authtype?cluster=${getCluster()}`, {}, false, false)
```

Comment:
If we had this coming from the cluster config, then maybe we wouldnt need it again.

Reply:
yes